public/app/core/components/NavBar/Next/NavBarMenu.tsx

```diff
@@ -83,9 +102,99 @@ const getStyles = (theme: GrafanaTheme2) => ({
     position: 'absolute',
     top: '43px',
     right: '0px',
+    transform: `translateX(50%)`,
+  }),
+  underlay: css({
+    position: 'fixed',
+    zIndex: theme.zIndex.modalBackdrop,
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+    backgroundColor: theme.components.overlay.background,
+    backdropFilter: 'blur(1px)',
   }),
 });
 
+const getAnimStyles = (theme: GrafanaTheme2, animationDuration: number) => {
+  const commonTransition = {
+    transitionProperty: 'width, background-color, opacity',
+    transitionDuration: `${animationDuration}ms`,
+    transitionTimingFunction: theme.transitions.easing.easeInOut,
+  };
+
+  const overlayTransition = {
+    ...commonTransition,
+    transitionProperty: 'width, background-color, box-shadow',
+  };
+
+  const underlayTransition = {
+    ...commonTransition,
+    transitionProperty: 'opacity',
+  };
+
+  const overlayOpen = {
+    backgroundColor: theme.colors.background.canvas,
+    boxShadow: theme.shadows.z3,
+    width: '300px',
+  };
+
+  const overlayClosed = {
+    backgroundColor: theme.colors.background.primary,
+    boxShadow: 'none',
+    width: theme.spacing(7),
+  };
+
+  const underlayOpen = {
+    opacity: 1,
+  };
+
+  const underlayClosed = {
+    opacity: 0,
+  };
+
+  return {
+    underlay: {
+      appear: css({
+        ...underlayClosed,
+      }),
+      appearActive: css({
+        ...underlayTransition,
+        ...underlayOpen,
+      }),
```

Comment:
can you pass these in as `css(underlayTransition, underlayOpen)`?

Reply:
yeah this is much nicer